### PR TITLE
SRE-738 Update Publish Action

### PR DESCRIPTION
**Pull request by d10zero**
Closes https://snapsheettech.atlassian.net/browse/SRE-738

Updating the ReadMe and the entrypoint.sh to not append 'v' to the github version and the tag. This removes the requirement of appending a 'v' to every tag.

This PR also adds a `prerelease` github action. Any time a new tag is created, a new prerelease version is created as well that can be used for testing.

## TEST
Use the ticker-ci-testing repo in this PR to test the pre-release version of this update: https://github.com/bodyshopbidsdotcom/tinker-ci-testing/pull/2

### CODE REVIEW

**Comment by whatisdot** [comment]
This aligns with the changes we discussed, which makes sense.

We need a testing plan for changes. Maybe we could set up a test repository? 🤔

File: entrypoint.sh [hunk 1]
```diff
@@ -14,7 +14,7 @@ fi
 echo "Verifying the version matches the gem version"
 VERSION_TAG=$(echo $GITHUB_REF | cut -d / -f 3)
 GEM_VERSION=$(ruby -e "require 'rubygems'; gemspec = Dir.entries('.').find { |file| file =~ /.*\.gemspec/ }; spec = Gem::Specification::load(gemspec); puts spec.version")
-if [[ $VERSION_TAG != "v$GEM_VERSION" ]]; then
+if [[ $VERSION_TAG != "$GEM_VERSION" ]]; then
```
Comment (whatisdot) [inline entrypoint.sh]: Are the double quotes around `$GEM_VERSION` still needed?

File: .github/workflows/pre-release.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  pre-release:
+    name: Pre Release
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Build & test
+        run: |
+          echo "done!"
+
+      - uses: marvinpinto/action-automatic-releases@latest
```
Comment (whatisdot) [inline .github/workflows/pre-release.yml]: We also should pin this to a specific version (currently `1.2.1`).

File: .github/workflows/pre-release.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        with:
+          repo_token: ${{ secrets.GITHUB_TOKEN }}
+          automatic_release_tag: "latest"
+          prerelease: true
+          title: "Development Build"
+          files: |
```
Comment (whatisdot) [inline .github/workflows/pre-release.yml]: The `prerelease` parameter defaults to `true`, and our build doesn't have any `.jar` files or a `LICENSE.txt` file. This raised an error in the Action output, but that didn't show as a failure. We should remove them.
Screen Shot 2021-06-24 at 12 57 41 PM

**Review by whatisdot — COMMENTED** [review]
Based on what I've read from the [marvinpinto/action-automatic-releases](https://github.com/marvinpinto/action-automatic-releases/) description, this automation does the following:
1. Whenever a new tag is pushed that matches the `*.*.*` pattern...
2. Build this GitHub Action, push the commits, and update the `latest` tag.

Teams can then test PRs by referencing the `latest` version for this Action.

It would be good to add a description of how to use it in a new "Development" section of the README.

**Review by whatisdot — APPROVED** [review]
The project dependency has been resolved.

**Review by deekom — APPROVED** [review]

**Pull request merged**